dimmer.el

```diff
@@ -202,11 +202,17 @@ wrong, then try HSL or RGB instead."
     (add-to-list
      'dimmer-prevent-dimming-predicates #'helm--alive-p)))
 
+(defun dimmer--hydra-p ()
+  "Check if current buffer is LV buffer."
+  (string= (buffer-name (current-buffer)) " *LV*"))
```

Comment:
So far the other uses of `dimmer-prevent-dimming-predicates` are functions that determine if the popup is active, even if the current buffer isn't itself the popup.  I just looked at the implementations for `helm--alive-p` and `which-key--popup-showing-p` to confirm I'm remembering this correctly.

I'd like to have a simple predicate function that determines if a hydra is active but doesn't depend on `current-buffer`. Looking at the [implementation of `hydra-pause-resume`](https://github.com/abo-abo/hydra/blob/bcc2d0eb5b7114688016c143397285d777dedac9/hydra.el#L1527-L1538) I think it might be enough to test if the value of `hydra-curr-map` is not nil. In other words, can you try the following?

```elisp
(defun dimmer--hydra-p ()
  "Check if any hydra is active."
  hydra-curr-map)
```

or, if you like, just use a lambda below

```elisp
    (add-to-list
      'dimmer-prevent-dimming-predicates (lambda () hydra-curr-map))
```

Reply:
There are packages that use `lv` but don't use `hydra` itself (e.g. [lsp-mode](https://github.com/emacs-lsp/lsp-mode/blob/e5d3fc5b02bc01aff5c5610923e3e18f0fe04774/lsp-mode.el#L45)). And `lv` is distributed separately [on melpa](https://melpa.org/#/lv). So it's not enough to check `hydra-curr-map` as it's present only in hydra buffers.

Reply:
We can check https://github.com/abo-abo/hydra/blob/bcc2d0eb5b7114688016c143397285d777dedac9/lv.el#L59 but not sure that I understood how `dimmer-prevent-dimming-predicates` works. If we just check this it means that while lv buffer is present dim state won't be changed at all?

Reply:
Ah, yes, got it. The strange thing is that when `lsp-mode` pops up `lv` message the main window becomes dimmed. It's the situation I wanted to prevent. But `lv` buffer can be active for a long time.

Reply:
Interesting, but I can't reproduce it anymore. So let's just fix `dimmer-exclusion-regexp-list` for now.

Reply:
Yes, to the question above about `dimmer-prevent-dimming-predicates` - it freezes the state of dimming until the predicate is false. It's really meant for transient popups like `which-key`. I'm not a hydra user and I frankly overlooked the fact that `lv` is released separately... thanks for that pointer.

I'll open a ticket to look into lv mode a little more closely.  Could also try something like
```elisp
(define dimmer--lv-p ()
  "Predicate to test if LV is active."
  (window-live-p lv-wnd))
```

Reply:
Actually I think `lv` shouldn't prevent buffer dimming. Lv message can be visible without being active and user can switch windows/buffers. You can easily test it with:
```elisp
(require 'lv)
(lv-message "aaaa")
```

Reply:
ok, thanks. I'll hold off on opening a ticket then. Let me know if you see further changes or improvements ...

Reply:
The problem with `lsp-mode` was not the `lv` but `lsp-ui` which creates child frame and after creation `post-command-hook` is executed with this buffer so main window is dimmed. The only way to fix it as I see to add check to `dimmer-prevent-dimming-predicates` for a current buffer - if it's a child lsp-ui frame then don't change dimming state. But again active lsp-ui frame shouldn't prevent dimming only if it's shown.